### PR TITLE
Add chapter on performance portable implementation.

### PR DESCRIPTION
**Pull request by markdewing**
The batched drivers are a major change, so it seems worthwhile to highlight in its own chapter.

Having a separate chapter also displays the title in the table of contents, which will make it more visible to users.

I move the 'batched drivers' section whole.  My intention is that this PR is for discussing whether adding the chapter is a good idea, and what it should be named.   Adjusting the content in the chapter will come later.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
laptop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- N/A. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: docs/performance_portable.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+The divergence came from the the fact that the CPU code path favors executing all the compute tasks within a step
+for one walker and then advance walker by walker. Multiple CPU threads process their own assigned walkers in parallel.
+In this way, walkers are not synchronized with each other and maximal throughout can be achieved on CPU.
+The GPU code path favors executing the same compute task over all the walkers together to maximize GPU throughput.
+This compute dispatch pattern minimizes the overhead of dispatching computation and host-device data transfer.
+However, the legacy GPU code path only leverages the OpenMP main host thread for handling
+all the interaction between the host and GPUs and limit the kernel dispatch capability.
+In brief, the CPU code path handles computation with a walker batch size of one and many batches
+while the GPU code path uses only one batch containing all the walkers.
+The new drivers that implement this flexible batching scheme are called "batched drivers".
```
Comment (ye-luo) [inline docs/performance_portable.rst:17]: "batched drivers" is only part of the performance portable design.
I think the next paragraph should remain in where it was. It was also needed in the old place for laying the knowledge for "transition guide". Once the next paragraph is moved back. Add the hyperlink here should be helpful.

**Review by ye-luo — APPROVED** [review]
This is an improvement. I think it is better to get this merged. Further updates can be in new PRs.
@prckent merge? Need non-ANL approval.

**Review by prckent — APPROVED** [review]
Thanks Mark. Sorry this took so long.

**Comment by prckent** [comment]
Test this please

**Pull request merged**